<launch>
  <arg name="bag_path" description="bagfile path"/>

  <arg name="map_path" description="point cloud and lanelet2 map directory path"/>
  <arg name="lanelet2_map_file" default="lanelet2_map.osm" description="lanelet2 map file name"/>
  <arg name="pointcloud_map_file" default="pointcloud_map.pcd" description="pointcloud map file name"/>

  <arg name="rviz" default="true" description="launch rviz"/>
  <arg name="rviz_config" default="$(find-pkg-share autoware_planning_data_analyzer)/rviz/autoware.rviz" description="rviz config"/>

  <arg name="use_sim_time" default="false" description="use_sim_time"/>
  <arg name="vehicle_model" default="sample_vehicle" description="vehicle model name"/>
  <arg name="sensor_model" default="sample_sensor_kit" description="sensor model name"/>

  <arg name="lanelet2_map_loader_param_path" default="$(find-pkg-share autoware_launch)/config/map/lanelet2_map_loader.param.yaml"/>
  <arg name="map_projection_loader_param_path" default="$(find-pkg-share autoware_launch)/config/map/map_projection_loader.param.yaml"/>

  <arg name="input/odometry" default="/localization/kinematic_state"/>
  <arg name="input/acceleration" default="/localization/acceleration"/>
  <arg name="input/trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="input/objects" default="/perception/object_recognition/objects"/>
  <arg name="input/map_topic" default="/map/vector_map"/>
  <arg name="input/route_topic" default="/planning/mission_planning/route"/>

  <group scoped="false">
    <include file="$(find-pkg-share global_parameter_loader)/launch/global_params.launch.py">
      <arg name="use_sim_time" value="$(var use_sim_time)"/>
      <arg name="vehicle_model" value="$(var vehicle_model)"/>
    </include>
  </group>

  <node_container pkg="rclcpp_components" exec="component_container" name="offline_evaluator_container" namespace="" output="screen">
    <composable_node pkg="autoware_offline_evaluation_tools" plugin="autoware::trajectory_selector::offline_evaluation_tools::OfflineEvaluatorNode" name="offline_evaluator">
      <param name="bag_path" value="$(var bag_path)"/>
      <param from="$(find-pkg-share autoware_offline_evaluation_tools)/config/offline_evaluation.param.yaml"/>
      <remap from="input/lanelet2_map" to="/map/vector_map"/>
    </composable_node>

    <composable_node pkg="glog_component" plugin="GlogComponent" name="glog_component" namespace=""/>
  </node_container>

  <group>
    <include file="$(find-pkg-share tier4_vehicle_launch)/launch/vehicle.launch.xml">
      <arg name="vehicle_model" value="$(var vehicle_model)"/>
      <arg name="sensor_model" value="$(var sensor_model)"/>
      <arg name="launch_vehicle_interface" value="false"/>
    </include>
  </group>

  <group>
    <include file="$(find-pkg-share autoware_launch)/launch/components/tier4_map_component.launch.xml"/>
  </group>

  <group if="$(var rviz)">
    <node pkg="rviz2" exec="rviz2" name="rviz2" output="screen" args="-d $(var rviz_config) -s $(find-pkg-share autoware_launch)/rviz/image/autoware.png" if="$(var rviz)" respawn="false"/>
  </group>
</launch>
